docs/mkdocs/docs/technical/contributing.md

```diff
@@ -0,0 +1,195 @@
+Setting up your OS dependencies
+===============================
+
+Linux
+-----
+The codebase and build system can work with any reasonably recent Linux distribution that ships at least
+GCC 8 (10+ recommended) and CMake 3.12 (3.18+ recommended).
+
+A development install of Python 3.6+ (with `libpython.a` or `.so` and full headers) is also necessary.
+(See [pybind11 configuration](#pybind11-configuration))
+
+Different distros/installs might have different names for their GCC and Python executables,
+so you might want to use the [CMake presets](#cmake-presets) feature.
+
+### Dependencies by distro
+| Distro | Versions reported to work | Packages |
+|---|---|---|
+| Ubuntu | 20.04, 22.04 | build-essential g++-10 libpcre3-dev libsasl2-dev libsodium-dev libkrb5-dev libcurl4-openssl-dev python3-dev |
+| Centos | 7 | devtoolset-10-gcc-c++ openssl-devel cyrus-sasl-devel devtoolset-10-libatomic-devel libcurl-devel python3-devel |
+
+ManyLinux Docker image
+----------------------
+ManyLinux(https://github.com/pypa/manylinux) captures the common denominator of many Linux distributions,
+so we can be confident that code compiled in this image runs everywhere.
+
+We [extend](https://github.com/man-group/ArcticDB/blob/master/build_tooling/build_many_linux_image.sh) it with
+additional dependencies for compilation purposes.
+You can build the same image locally by running that script.
+We're looking into hosting the image in a public location.
+
+Windows
+-------
+We recommend using Visual Studio 2022 (or later) to install the compiler (MSVC v142 or newer) and tools
+(Windows SDK, CMake, Python).
+
+The Python that comes with Visual Studio is sufficient for creating release builds, but for debug builds, you will have
+to separately download from [Python.org](https://www.python.org/downloads/windows/).
```

Comment:
add a note to download a debug version of Python?

Reply:
I did mention that I'll leave this section mostly out for you to contribute :P